lib/OpenQA/Worker/Jobs.pm

```diff
@@ -524,7 +524,11 @@ sub start_job {
 
     # update settings with worker-specific stuff
     copy_job_settings($job, $worker_settings);
-
+    unlink("$pooldir/autoinst-log.txt");
```

Comment:
are we sure we're autodie free here? :)

Reply:
If the files are not present, the worker will keep going on with it's life unless it cannot write on it and die horribly :) causing worker IOloop to die

Reply:
So far when I wanted the autoinst-log.txt to be cleaned I overwrote the content with 0 bytes and deleted the ulogs directory before starting a new test to prevent lengthy uploads of all different kind of logfiles which one normally does not need for test development. If someone wants to preserve the logfile then this would make it quite hard.

I vote for keeping the old behaviour because "no-cleanup" is clear as of now: No cleanup (of anything). Removing the content of *some* files but not of others is … inconsistent.

Reply:
well, --no-cleanup is documented as not doing anything *after* job. It doesn't say anything about not cleaning up on start - but it would be indeed quite a suprise if we started erasing images :)

I just have a hard time finding a use case of keeping the logs - it's debug output of the test results uploaded. We just happen to append to the file.

Reply:
Problem with --no-cleanup is that if a worker picks up a job, next time a job runs... the person that is interested in the job run, will see double logs (@SergioAtSUSE and @asmorodskyi) for example :). The log files are uploaded to the webUI and then become irrelevant, so are the images and the rest of test results, except the disk file, and perhaps the job.json file that mimght be of some interest.

We can add a switch for every part:
` --keep-images
  --keep-thumbnails
  --keep-testresults
  --keep-video
` come to mind. 

@coolo so, unlink or warn('file not found')?